src/node/rpc/handlerregistry.h

```diff
@@ -190,23 +210,28 @@ namespace ccf
 
     virtual void tick(std::chrono::milliseconds elapsed, size_t tx_count) {}
 
-    virtual std::optional<CallerId> valid_caller(
+    bool has_certs()
+    {
+      return certs != nullptr;
+    }
+
+    virtual CallerId get_caller_id(
```

Comment:
This function (renamed from `valid_caller()`) now always returns a `CallerId` (`INVALID_ID` if the certs table is not set or if the RPC is anonymous). This simplifies the logic in `frontend.h` quite a bit.